File: gdscript/src/main/java/gdscript/psi/impl/GdPatternListImpl.java
package gdscript.psi.impl;

import com.intellij.extapi.psi.ASTWrapperPsiElement;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import com.intellij.psi.util.PsiTreeUtil;
import gdscript.psi.GdPattern;
import gdscript.psi.GdPatternList;
import gdscript.psi.GdVisitor;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class GdPatternListImpl extends ASTWrapperPsiElement implements GdPatternList {

  public GdPatternListImpl(@NotNull ASTNode node) {
    super(node);
  }

  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitPatternList(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

  @Override
  @NotNull
  public List<GdPattern> getPatternList() {
    return PsiTreeUtil.getChildrenOfTypeAsList(this, GdPattern.class);
  }

}

File: gdscript/src/main/java/gdscript/psi/impl/GdPatternListImpl.kt
package gdscript.psi.impl

import com.intellij.extapi.psi.ASTWrapperPsiElement
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import com.intellij.psi.util.PsiTreeUtil
import gdscript.psi.GdPattern
import gdscript.psi.GdPatternList
import gdscript.psi.GdVisitor

class GdPatternListImpl(node: ASTNode) : ASTWrapperPsiElement(node), GdPatternList {
    fun accept(visitor: GdVisitor) {
        visitor.visitPatternList(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }

    override val patternList: List<GdPattern>
        get() = PsiTreeUtil.getChildrenOfTypeAsList(this, GdPattern::class.java)
}
